engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -271,6 +270,32 @@ Object doFunctionalDispatchUncachedSymbol(
     return invokeFunctionNode.execute(function, frame, state, arguments);
   }
 
+  private PanicException methodNotFound(UnresolvedSymbol symbol, Object self) throws PanicException {
+    var cause = onBoundary ? UnknownIdentifierException.create(symbol.getName()) : null;
+    var payload = EnsoContext.get(this).getBuiltins().error().makeNoSuchMethod(self, symbol);
+    throw new PanicException(payload, cause, this);
+  }
+
+  @Specialization
+  Object doMultiValue(
+      VirtualFrame frame,
+      State state,
+      UnresolvedSymbol symbol,
+      EnsoMultiValue self,
+      Object[] arguments,
+      @Shared("methodResolverNode") @Cached MethodResolverNode methodResolverNode
+  ) {
+    var fnAndType = self.resolveSymbol(methodResolverNode, symbol);
+    if (fnAndType != null) {
+      var unwrapSelf = self.castTo(fnAndType.getRight());
+      assert unwrapSelf != null;
+      assert arguments[0] == self;
+      arguments[0] = unwrapSelf;
+      return execute(frame, state, symbol, unwrapSelf, arguments);
```

Comment:
No need to resolve the Function once more - it is already resolved in `fnAndType`.

```suggestion
return invokeFunctionNode.execute(fnAndType.getLeft(), frame, state, arguments);
```

Reply:
That's very good observation. Thank you.